src/cleanfigure.m

```diff
@@ -367,6 +368,259 @@ function coarsenLine(meta, handle, minimumPointsDistance)
   return;
 end
 % =========================================================================
+function simplifyLine(meta, handle, targetResolution)
+  % Reduce the number of data points in the line handle by removing
+  % points which add features with area smaller than 1/4 of a pixel at the
+  % target resolution. 
+  %
+  % targetResolution is in format 'WxH@Res', eg. '15x9@600' for a 15cm by 9cm
+  % figure at 600 ppcm.
+  % 
+  % Use targetResolution = 'off' to disable line simplification
+
+  % Extract the data from the current line handle.
+  xData = get(handle, 'XData');
+  yData = get(handle, 'YData');
+  zData = get(handle, 'ZData');
+
+  if ~isempty(zData)
+    % Don't simplify 3d plots
+    return;
+  end
```

Comment:
Regarding 3D: let's indeed put a TODO. 
Probably one of the reasons why this hasn't been addressed (in our old code and here), is because in 3D there are more intricate things that can be done improperly than in 2D:
- we need to reduce the number of points (e.g. with such an algorithm)
- the matter of perspective / view angle comes into play to determine whether a feature is distinguishable or not
- the data formats used by the different objects is more complicated (e.g. X/Y vectors or possibly meshgridded matrices and Z a matrix)
- the simplified data matrix needs to comply to the same constraints: I expect that we can only remove a complete row or column of the Z matrix in general, except maybe for line plots (which is a direct generalization of this code, but with 3D points).